src/application/deployer.go

```diff
@@ -19,7 +19,13 @@ func StartDeploy(deployConfig types.DeployConfig) error {
 	}
 
 	deployConfig.LogChannel <- fmt.Sprintf("Building %s %s...\n\n", deployConfig.AppConfig.Name, deployConfig.AppConfig.Version)
-	initializer, err := NewInitializer(deployConfig.AppConfig, deployConfig.LogChannel, "exo-deploy", deployConfig.AppDir, deployConfig.HomeDir)
+	initializer, err := NewInitializer(
+		deployConfig.AppConfig,
+		deployConfig.LogChannel,
+		"exo-deploy",
+		deployConfig.AppDir,
+		deployConfig.HomeDir,
+		deployConfig.DockerComposeProjectName)
```

Comment:
Thoughts on adding a `GetDockerComposeProjectName` to the `AppConfig` struct so we don't need this extra parameter to initializer / runner / deployer?

Reply:
I liked created the project name at the highest level because that ensures we're creating it in one place and passing the same name down, as opposed to each of `initializer` / `runnner` creating their own for example. If we run `docker-compose -p app1 initializer build` then `docker-compose -p app2 initializer up` then docker compose will think we're running two different projects and it won't behave as expected. I already ran into some problems with this writing tests.

Also, `deployer` and `tester` both use `initializer`, but they require different project names so that needs to be passed down instead of generated inside initializer.

Reply:
If we wanted to clean it up a little bit we could create the project name in `application/deployer` and `application/tester`, but for the runner we'd still have to create it in `cmd/run`, b/c that's where both `initializer` and `runner` are created.